### PR TITLE
Remove legacy generate and bundle commands

### PR DESCRIPTION
**Pull request by hasbro17**
**Description of the change:**
Removed the following legacy cmds:
- `generate csv`
- `generate packagemanifests`
- `generate bundle`
- `bundle create`
- `bundle validate`

Also removed associated scaffolding packages for these commands in `internal/generate/olm-catalog`.

**Motivation for the change:**

From #3327 : Prepare SDK CLI for `1.0.0` release

**TODO:**

- [x] Wait for #3406 to merge to remove errors with legacy `run packagemanifests` cmd using `internal/generate/olm-catalog`.
- [x] Remove legacy CLI docs from the website
- [x] ~~Remove legacy OLM integration docs_ https://sdk.operatorframework.io/docs/olm-integration/legacy/~~
- [x] Remove legacy `generate bundle` cmd
- [x] Remove entire legacy `bundle <create/validate>` cmd

### CODE REVIEW

File: cmd/operator-sdk/bundle/cmd.go [hunk 1]
```diff
@@ -51,7 +51,6 @@ More information about the integration with OLM via SDK:
 https://sdk.operatorframework.io/docs/olm-integration/legacy
 `
 	cmd.AddCommand(
```
Comment (estroz) [inline cmd/operator-sdk/bundle/cmd.go:53]: You can remove the entire `bundle` subcommand for legacy projects.

File: internal/generate/olm-catalog/descriptor/descriptor.go [hunk 2]
```diff
@@ -1,261 +0,0 @@
-// Copyright 2019 The Operator-SDK Authors
```
Comment (estroz) [inline internal/generate/olm-catalog/descriptor/descriptor.go:1]: This package still needs to stick around for the new CSV bases to work. #2792 should be updated to remove these. You can just leave this package in `internal/generate/olm-catalog/descriptor` for now.

File: cmd/operator-sdk/bundle/cmd.go [hunk 3]
```diff
@@ -44,19 +44,6 @@ https://github.com/operator-framework/operator-registry/blob/master/docs/design/
...
-https://sdk.operatorframework.io/docs/olm-integration/legacy
-`
-	cmd.AddCommand(
-		newCreateCmd(),
-		newValidateCmdLegacy(),
-	)
-	return cmd
-}
-
 func NewCmd() *cobra.Command {
```
Comment (estroz) [inline cmd/operator-sdk/bundle/cmd.go:47]: You can combine `newCmd` and `NewCmd` now that the legacy wrapper is gone.

File: cmd/operator-sdk/generate/cmd.go [hunk 4]
```diff
@@ -41,14 +41,3 @@ func NewCmd() *cobra.Command {
 	)
 	return cmd
```
Comment (estroz) [inline cmd/operator-sdk/generate/cmd.go:42]: You can combine `newCmd` and `NewCmd` now that the legacy wrapper is gone.

**Comment by hasbro17** [comment]
Thanks for the quick reviews everyone.

**Pull request merged**